docs/sql-programming-guide.md

```diff
@@ -581,6 +581,15 @@ Configuration of Parquet can be done using the `setConf` method on SQLContext or
   </td>
 </tr>
 <tr>
+  <td><code>spark.sql.parquet.int96AsTimestamp</code></td>
+  <td>true</td>
+  <td>
+    Some Parquet-producing systems, in particular Impala, store Timestamp into INT96. Spark would also
+    store Timestamp as INT96 because we need to avoid precision lost of the nanoseconds field. This
+    flag tells Spark SQL to interpret INT96 data as a timestamp to provide compatibility with these systems.
+  </td>
+</tr>
```

Comment:
Do they?  As far as I can tell the parquet spec does not have a nanosecond
precision timestamp type.
On Jan 20, 2015 12:26 AM, "Cheng Lian" <notifications@github.com> wrote:

> In docs/sql-programming-guide.md
> <https://github.com/apache/spark/pull/3820#discussion-diff-23207993>:
>
> > @@ -581,6 +581,15 @@ Configuration of Parquet can be done using the `setConf` method on SQLContext or
> >    </td>
> >  </tr>
> >  <tr>
> > +  <td><code>spark.sql.parquet.int96AsTimestamp</code></td>
> > +  <td>true</td>
> > +  <td>
> > +    Some Parquet-producing systems, in particular Impala, store Timestamp into INT96. Spark would also
> > +    store Timestamp as INT96 because we need to avoid precision lost of the nanoseconds field. This
> > +    flag tells Spark SQL to interpret INT96 data as a timestamp to provide compatibility with these systems.
> > +  </td>
> > +</tr>
>
> I'm probably OK with this property. Just wanna ask doesn't Impala store
> original type information (TIMESTAMP) together in Parquet metainfo?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/spark/pull/3820/files#r23207993>.
>